client/asset/dcr/spv.go

```diff
@@ -875,79 +921,101 @@ func (w *spvWallet) PurchaseTickets(ctx context.Context, n int, vspHost, vspPubK
 	if err != nil {
 		return nil, err
 	}
-	hashes := res.TicketHashes
-	hashStrs := make([]string, len(hashes))
-	for i := range hashes {
-		hashStrs[i] = hashes[i].String()
+
+	tickets := make([]*asset.Ticket, len(res.TicketHashes))
+	for i, h := range res.TicketHashes {
+		ticketSummary, hdr, err := w.dcrWallet.GetTicketInfo(ctx, h)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching info for new ticket")
+		}
+		ticket := ticketSummaryToAssetTicket(ticketSummary, hdr, w.log)
+		if ticket == nil {
+			return nil, fmt.Errorf("invalid ticket summary for %s", h)
+		}
+		tickets[i] = ticket
 	}
-	return hashStrs, err
+	return tickets, err
 }
 
+const (
+	upperHeightMempool   = -1
+	lowerHeightAutomatic = -1
+	pageSizeUnlimited    = 0
+)
+
 // Tickets returns current active tickets.
 func (w *spvWallet) Tickets(ctx context.Context) ([]*asset.Ticket, error) {
-	return w.ticketsInRange(ctx, 0, 0)
+	return w.ticketsInRange(ctx, lowerHeightAutomatic, upperHeightMempool, pageSizeUnlimited, 0)
 }
 
-func (w *spvWallet) ticketsInRange(ctx context.Context, fromHeight, toHeight int32) ([]*asset.Ticket, error) {
-	params := w.ChainParams()
+var _ ticketPager = (*spvWallet)(nil)
 
-	tickets := make([]*asset.Ticket, 0)
+func (w *spvWallet) TicketPage(ctx context.Context, scanStart int32, n, skipN int) ([]*asset.Ticket, error) {
+	if scanStart == -1 {
+		_, scanStart = w.MainChainTip(ctx)
+	}
+	return w.ticketsInRange(ctx, 0, scanStart, n, skipN)
+}
+
+func (w *spvWallet) ticketsInRange(ctx context.Context, lowerHeight, upperHeight int32, maxN, skipN /* 0 = mempool */ int) ([]*asset.Ticket, error) {
+	p := w.chainParams
+	const requiredConfs = 6 + 2
+	var startBlock, endBlock *wallet.BlockIdentifier // null endBlock goes through mempool
+	// If mempool is included, there is no way to scan backwards.
+	includeMempool := upperHeight == upperHeightMempool
+	if includeMempool {
+		_, upperHeight = w.MainChainTip(ctx)
+	} else {
+		endBlock = wallet.NewBlockIdentifierFromHeight(upperHeight)
+	}
+	if lowerHeight == lowerHeightAutomatic {
+		bn := upperHeight - int32(p.TicketExpiry+uint32(p.TicketMaturity)-requiredConfs)
```

Comment:
What's the reason for subtracting `requiredConfs` and why is it `6 + 2`?

Reply:
It's used in `ForUnspentUnexpiredTickets` without comment, but yes, I agree it seems unneeded. I guess I'll just remove it.